services/alertmanager/alertmanager.go

```diff
@@ -181,6 +195,68 @@ func (svc *Service) updateConfiguration(ctx context.Context) {
 	svc.l.Infof("%s created", alertmanagerConfigPath)
 }
 
+// collectRuleTemplates collects IA rule templates from various sources like
+// templates shipped with PMM and defined by the users.
+func (svc *Service) collectRuleTemplates() {
+	shippedFilePaths, err := filepath.Glob(svc.shippedRuleTemplatePath)
+	if err != nil {
+		svc.l.Errorf("Failed to get paths of template files shipped with PMM: %s.", err)
+		return // keep previously loaded rules
+	}
+
+	for _, path := range shippedFilePaths {
+		rules, err := svc.loadRuleTemplates(path)
+		if err != nil {
+			svc.l.Errorf("Failed to load shipped rule template file: %s, reason: %s.", path, err)
+			return // keep previously loaded rules
+		}
+		svc.updateRules(rules)
+	}
+
+	userDefinedFilePaths, err := filepath.Glob(svc.userDefinedRuleTemplatePath)
+	if err != nil {
+		svc.l.Errorf("Failed to get paths of user-defined template files: %s.", err)
+		return // keep previously loaded rules
+	}
+
+	for _, path := range userDefinedFilePaths {
+		rules, err := svc.loadRuleTemplates(path)
+		if err != nil {
+			svc.l.Errorf("Failed to load user-defined rule template file: %s, reason: %s.", path, err)
+			return // keep previously loaded rules
+		}
+		svc.updateRules(rules)
+	}
+
+	// TODO download templates from SAAS.
+}
+
+// loadRuleTemplates parses IA rule template files.
+func (svc *Service) loadRuleTemplates(file string) ([]saas.Rule, error) {
+	data, err := ioutil.ReadFile(file) //nolint:gosec
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read test rule template file")
+	}
+
+	// be strict about local files
+	params := &saas.ParseParams{
+		DisallowUnknownFields: true,
+		DisallowInvalidRules:  true,
+	}
+	rules, err := saas.Parse(bytes.NewReader(data), params)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse test rule template file")
+	}
+
+	return rules, nil
+}
+
+func (svc *Service) updateRules(rules []saas.Rule) {
+	for _, r := range rules {
+		svc.rules[r.Name] = r
```

Comment:
So you only add to that map, and never clean it? So once loaded rule will be present in map forever?